src/EventStore.Core/LogAbstraction/INameIndexExtensions.cs

```diff
@@ -30,5 +30,33 @@ public static class INameIndexExtensions {
 
 			return preExisting;
 		}
+		
+		public static bool GetOrReserveEventType<TStreamId>(
+			this INameIndex<TStreamId> eventTypeIndex,
+			IRecordFactory<TStreamId> recordFactory,
+			string eventType,
+			long logPosition,
+			out TStreamId eventTypeId,
+			out IPrepareLogRecord<TStreamId> eventTypeRecord) {
+
+			var preExisting = eventTypeIndex.GetOrReserve(eventType, out eventTypeId, out var addedNumber, out var addedName);
+
+			var appendNewEventType = recordFactory.ExplicitStreamCreation && !preExisting;
```

Comment:
@pvanbuijtene @timothycoleman do we want a separate property for event type creation in the record factory ? (e.g ExplicitEventTypeCreation) or is it fine?

Reply:
i'll add one for event type creation for neatness